onboarding/source/amb_light_service.c

```diff
@@ -18,22 +29,113 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if (lightServiceTaskHandle == NULL) {
+        // Create light service task
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
+    }
+
+    // Check if task was created successfully
+    if (xReturned == pdFAIL) {
+        return 0;
+    }
 
+    if (lightServiceQueue == NULL) {
+        // Create light service queue
+        lightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_LENGTH,
+                                        LIGHT_SERVICE_QUEUE_ITEM_SIZE);
+    }
+
+    // Check if queue was created successfully
+    if (lightServiceQueue == NULL) {
+        return 0;
+    }
     /* USER CODE END */
     return 1;
 }
 
+static void adcStartConversion_selChn(adcBASE_t *adc, unsigned channel, unsigned fifo_size, unsigned group) {
```

Comment:
To improve clarity, change all `unsigned` arguments/variables to the appropriate fixed-size integer type (like `uint8_t` or `uint32_t`)